BMS Workspace-git/bms-1227-fw/Phantom_drivers/rtos/include/queue_common.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * queue_common.h
+ * Typedefs for structures used in agent, actor-type software architecture
+ *
+ *  Created on: May 13, 2022
+ *      Author: dlutz
+ */
+
+#ifndef PHANTOM_DRIVERS_RTOS_INCLUDE_QUEUE_COMMON_H_
+#define PHANTOM_DRIVERS_RTOS_INCLUDE_QUEUE_COMMON_H_
+
+#include "stdio.h"
+#include "stdint.h"
+#include "FreeRTOS.h"
+#include "os_task.h"
+#include "os_queue.h"
+
+/* Agent Message Structure */
+typedef struct AgentMessage {
+    uint8_t id;  // agent identifier
+    uint8_t msg; // data from agent
+} AgentMessage_t;
+
+// Data structure for messages for this agent:
+typedef struct bmsSlaveMsg_t {
+    /* Input Data */
+    AgentMessage_t data;
+    uint8_t agent_state;
+    uint8_t actor_state;
+
+    /* Output Data */
+    uint8_t tx_data;
+
+} bmsSlaveMsg_t;
+
+typedef struct QueueArr_t {
+    QueueHandle_t tx;
+    QueueHandle_t rx;
+} QueueArr_t;

```

Comment:
Hesitant about enforcing one tx and rx queue...it's fine for this PR just tagging it for future changes

Reply:
It'd be easy enough now, before this moves, to move to a different naming scheme - maybe "tx_one", "rx_one" or something else if Phantom has a set naming scheme somewhere - so that if we wanted to expand the struct later on we could keep a consistent naming scheme while adding new queues.

Reply:
This is a struct definition. We still can name the instances and make it more readable that way. So it should be fine how it is